plugins/inputs/mongodb/sample.conf

```diff
@@ -36,3 +36,9 @@
   # tls_key = "/etc/telegraf/key.pem"
   ## Use TLS but skip chain & host verification
   # insecure_skip_verify = false
+
+  ## When true, telegraf will not return an error on start if one of
+  ## the mongodb targets is unreachable
+  ## Note that while running, telegraf will also avoid data gathering for
+  ## unreachable mongo endpoints
+  # ignore_unreachable_hosts = false
```

Comment:
Could we change the name and the type of the config setting? It sounds like this fix only ignores server errors temporarily, but this name implies to me that it will ignore servers completely. I think from reading the code that it retries connecting to the server each gather and just ignores connection errors.

Earlier in the config we call the hosts "servers". It would be more clear to users if we continued using the same term here.

Also, using a boolean setting doesn't give any room to add other ways to handle the problem to be added in the future. If we use a string setting, we could add any number of future approaches.

```toml
disconnected_servers = "retry"
```

Then if someone wanted to do something else when a server disconnects (like log or remove from the list or whatever) there will already be a way to configure it.

Reply:
Will do it for the naming hamonization.

On the "behavior" proposal, I was thinking about this at first but wasn't sure on the way to implement it : 

- What kind of strictness do we want, what happens if the option is not one of the accepted values ?
- Where the validation code should be place ? I didn't find something similar in the code base, do I have to make the check logic inside the plugin ?

Reply:
The init function does the validation of the config, there is the internal `choice.Check` method.